r/R/atac_utils.R

```diff
@@ -379,30 +379,260 @@ call_peaks_tile <- function(fragments, chromosome_sizes, cell_groups = rep.int("
 #' number insertions at each basepair for each group listed in `cell_groups`.
 #'
 #' @param fragments IterableFragments object
-#' @param path Path(s) to save bedgraph to, optionally ending in ".gz" to add gzip compression. If `cell_groups` is provided,
+#' @param path (character vector) Path(s) to save bedgraph to, optionally ending in ".gz" to add gzip compression. If `cell_groups` is provided,
 #'   `path` must be a named character vector, with one name for each level in `cell_groups`
-#' @param insertion_mode Which fragment ends to use for insertion counts calculation. One of "both", "start_only", or "end_only"
+#' @param insertion_mode (string) Which fragment ends to use for coverage calculation. One of "both", "start_only", or "end_only"
 #' @inheritParams footprint
 #' @export
-write_insertion_bedgraph <- function(fragments, path, cell_groups = NULL, insertion_mode=c("both", "start_only", "end_only")) {
+write_insertion_bedgraph <- function(fragments, path, cell_groups = rlang::rep_along(cellNames(fragments), "all"), insertion_mode=c("both", "start_only", "end_only")) {
   assert_is(fragments, "IterableFragments")
   assert_is_character(path)
+  assert_is(cell_groups, c("character", "factor"))
   insertion_mode <- match.arg(insertion_mode)
+  path_names <- names(path)
+  path <- suppressWarnings(normalizePath(path))
+  names(path) <- path_names
+  cell_groups <- as.factor(cell_groups)
+  assert_len(path, length(levels(cell_groups)))
   
-  if (is.null(cell_groups)) {
-    cell_groups <- rep.int(0L, length(cellNames(fragments)))
-    assert_len(path, 1)
+  if (length(levels(cell_groups)) == 1) {
+    names(path) <- "all"
   } else {
-    assert_is(cell_groups, c("character", "factor"))
     assert_len(cell_groups, length(cellNames(fragments)))
-    cell_groups <- as.factor(cell_groups)
     assert_has_names(path, levels(cell_groups))
     path <- path[levels(cell_groups)]
     cell_groups <- as.integer(cell_groups) - 1L
   }
   write_insertion_bedgraph_cpp(iterate_fragments(fragments), cell_groups, path, insertion_mode)
 }
 
+#' Create bed files from fragments split by cell group.
+#' @param path (character vector) Path to save bed files. If `cell_groups` is provided, this must be a character vector with one name for each level in `cell_groups` 
+#' Else, this must be a character vector of length 1.
+#' @param cell_groups (character vector or factor) Cluster assignments for each cell. If is null, all cells are treated as one group.
```

Comment:
```suggestion
#' @param cell_groups (character vector or factor) Cluster assignments for each cell.
```
Looks like we no longer have the `NULL` option so this can come out of the docs